frontends/web/src/components/devices/bitbox02/bitbox02.tsx

```diff
@@ -34,43 +39,117 @@ interface Props {
+                        case 'statusChanged':
+                            this.onStatusChanged();
+                            break;
+                    }
+                    break;
+            }
+        });
+    }
+
+    private onChannelHashChanged = () => {
+        apiGet('devices/bitbox02/' + this.props.deviceID + '/channel-hash').then(({ hash, deviceVerified }) => {
+            this.setState({ hash, deviceVerified });
+        });
+    }
+
+    private onStatusChanged = () => {
+        apiGet('devices/bitbox02/' + this.props.deviceID + '/status').then(status => {
+            this.setState({ status });
         });
     }
 
     public componentWillUnmount() {
         this.unsubscribe();
     }
 
-    private abort = () => {
-        this.setState({ verified: true });
+    private channelVerify = ok => {
+        apiPost('devices/bitbox02/' + this.props.deviceID + '/channel-hash-verify', ok);
+    }
+
+    private setPassword = () => {
+        this.setState({ settingPassword: true });
+        apiPost('devices/bitbox02/' + this.props.deviceID + '/set-password').then(({ success }) => {
+            if (!success) {
+                alertUser('pw did not match, try again');
+            }
+            this.setState({ settingPassword: false });
+        });
+    }
+
+    private createBackup = () => {
+        this.setState({ creatingBackup: true });
+        apiPost('devices/bitbox02/' + this.props.deviceID + '/create-backup').then(({ success }) => {
+            if (!success) {
+                alertUser('creating backup failed, try again');
+            }
+            this.setState({ creatingBackup: false });
+        });
     }
 
-    public render({ deviceID }: RenderableProps<Props>, { hash, verified }: State) {
-        return (
-            <div>
-                <span>Hello BitBox02</span>
-                <RandomNumber apiPrefix={'devices/bitbox02/' + deviceID} />
-                <DeviceInfo apiPrefix={'devices/bitbox02/' + deviceID} />
-                <SetDeviceName apiPrefix={'devices/bitbox02/' + deviceID} />
-                {
-                    !verified ? (
-                        <Dialog onClose={this.abort}>
-                            <p>Verify your BitBox</p>
-                            <p>{hash}</p>
-                            <Button primary onClick={this.abort}>Correct</Button>
-                        </Dialog>
-                    ) : null
-                }
-            </div>
-        );
+    public render({ deviceID }: RenderableProps<Props>, { hash, deviceVerified, status, settingPassword, creatingBackup }: State) {
+        switch (status) {
+            case 'unpaired':
+                return (
+                    <Dialog onClose={() => this.channelVerify(false)}>
+                        <p>Verify your BitBox</p>
+                        <p>{hash}</p>
+                        <Button primary onClick={() => this.channelVerify(true)} disabled={!deviceVerified}>Correct</Button>
```

Comment:
I was going to put this on the firmware PR, but that one is merged now :)
In `static void _confirm(component_t* component)`, in pairing.c, the screen is popped from the device before user can press confirm on the app.
When confirming on app, user can't see the code on the device anymore. The app could potentially change the code the user is confirming with no way to compare anymore.